scripts/tests/cross-test-ci-all.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+set -x
+
+# Test current versions against gatewayd v0.2
+export FM_GATEWAYD_BASE_EXECUTABLE="nix run git+https://github.com/fedimint/fedimint.git?ref=releases/v0.2#gatewayd --"
```

Comment:
There are couple of problems with this:

* some extra stuff gets printed on stderr which can confuse some tools and pollute the logs
* the slowdown happens somewhere deep inside so it might be hard to tell if the whole thing hanged or just building

A:

```
>&2 echo "Compiling old binaries..."
nix build git+https://github.com/fedimint/fedimint.git?ref=releases/v0.2#gatewayd
PATH="$(readlink result)/bin/:$PATH"
```

should achieve the same thing without the downsides, and also doesn't require parsing shell commands as arguments in devimint.

Reply:
@bradleystachurski We've discussed adding old releases to inputs, but I guess the above approach has a nice property that it's on-demand, not requiring building old binaries upfront, so no need for extra flake inputs and/or separate dev shells.

Reply:
BTW `nix build 'github:fedimint/fedimint?branch=releases/0.2#gatewayd'` is shorter

Reply:
Since the `releases/v0.2` branch merges backport PRs, it's a bit misleading. Right now `relesaes/v0.2` (https://github.com/fedimint/fedimint/commit/51d09c76b8a15615114d4f2c2663c738d8ff3135) is using [v0.2.1-rc2](https://github.com/fedimint/fedimint/releases/tag/v0.2.1-rc2) as the base.

I think it's more clear to use the explicit tag we want as a base and also allows for a shorter nix command
```
nix build 'github:fedimint/fedimint/v0.2.0#gatewayd'
```

Reply:
Seems to me we should just override binaries by writing to the PATH

Reply:
@dpc re: https://github.com/fedimint/fedimint/pull/3982/files#r1433538187

How would we handle the binaries with a same name and different version using this approach?

Would it make sense to export the path for the specific binary instead of prepending to the PATH?

```
nix build 'github:fedimint/fedimint/v0.2.0#gatewayd'
export FM_GATEWAYD_BASE_EXECUTABLE="$(readlink result)/bin/gatewayd"

Reply:
Instead of `readlink` we could use `nix build .#… --print-out-paths`

Reply:
You all all right! :)